tests/steps/test_docstring.py

```diff
@@ -0,0 +1,144 @@
+import textwrap
+
+from src.pytest_bdd.utils import collect_dumped_objects
+
+
+def test_steps_with_docstrings(pytester):
+    pytester.makefile(
+        ".feature",
+        docstring=textwrap.dedent(
+            '''
+            Feature: Docstring
+
+              Scenario: Step with plain docstring as multiline step
+                Given a step has a docstring
+                """
+                This is a given docstring
+                """
+
+                When a step provides a docstring with lower indentation
+                """
+            This is a when docstring
+                """
+
+                And this step has no docstring
+
+                Then this step has a greater indentation
+                """
+                        This is a then docstring
+                """
+            '''
+        ),
+    )
+
+    pytester.makeconftest(
+        textwrap.dedent(
+            r"""
+        from pytest_bdd import given, when, then, parsers
+        from pytest_bdd.utils import dump_obj
+        import re
+
+
+        @given("a step has a docstring")
+        def _(docstring):
+            given_docstring = docstring.strip()
+            dump_obj(given_docstring)
+
+
+        @when("a step provides a docstring with lower indentation")
+        def _(docstring):
+            when_docstring = docstring.strip()
+            dump_obj(when_docstring)
+
+
+        @when("this step has no docstring")
+        def _():
+            pass
+
+
+        @then("this step has a greater indentation")
+        def _(docstring):
+            then_docstring = docstring.strip()
+            dump_obj(then_docstring)
+            """
+        )
+    )
+
+    pytester.makepyfile(
+        textwrap.dedent(
+            """\
+            from pytest_bdd import scenario
+
+            @scenario("docstring.feature", "Step with plain docstring as multiline step")
+            def test_docstring():
+                pass
+            """
+        )
+    )
+
+    result = pytester.runpytest("-s")
+    result.assert_outcomes(passed=1)
+
+    docstrings = collect_dumped_objects(result)
+    assert docstrings[0] == "This is a given docstring"
+    assert docstrings[1] == "This is a when docstring"
+    assert docstrings[2] == "This is a then docstring"
+
+
+def test_steps_with_missing_docstring(pytester):
```

Comment:
can we also add a test that if the step impl doesn't have the `docstring` param, a (human readable) error is raised?

Reply:
This is incorrect (if I am understanding correctly).  If the step has a gherkin docstring in the feature file, the step def doesn't necessarily need to use the docstring arg to retrieve it.  If I miss it out in the step def, it ignores the docstring

Reply:
oh, is that the same for the `datatable` arg?
We definitely need a test to assert the behaviour, but I'm happy if the code detect that the step impl does not require that arg, and avoids passing it.

We should though make sure that it is passed if the step impl has a `**kwargs` param

Reply:
I'll add a test for datatable and I'll check the last part

Reply:
Hmmm I see, the `get_args` function only returns the declared args...
I'd say then to fix that behaviour in a next PR, and undo the last change you did for now if you don't mind

Reply:
Done :)